spec/utils.tsx

```diff
@@ -20,3 +22,12 @@ export function createWrapper(nostoContext: NostoContextType) {
     return <NostoContext.Provider value={nostoContext}>{children}</NostoContext.Provider>
   }
 }
+
+export async function waitForRecommendations(num: number) {
```

Comment:
Did you consider the option of using a callback as a method parameter instead of restricting the  assertion to 
```js
expect(screen.getAllByTestId("recommendation")).toHaveLength(num)
```

Reply:
how would the call look like from the usage side?

Reply:
It could be something like 

```js
waitForRecommendations(() => {
    expect(screen.getAllByTestId("recommendation")).toHaveLength(2)
    .... // other assertions
})
```
This way, this reusable method will be more generic

Reply:
true, it would be a more generic utility, but we would only save the timeout part

Reply:
That's fine IMO because we are using the same timeout across all assertions. But, on the other hand, we may not always restrict this to a single assertion. If this reusable method is placed inside a test, then it would have made more sense.

Reply:
Also, make the `WAIT_FOR_TIMEOUT` a local constant instead of exporting it?